lib/facter/facts/solaris/dmi/product/name.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Facts
+  module Solaris
+    module Dmi
+      module Product
+        class Name
+          FACT_NAME = 'dmi.product.name'
+          ALIASES = 'productname'
+
+          def call_the_resolver
+            fact_value = nil
+            if isa == 'i386'
+              fact_value = Facter::Resolvers::Solaris::Dmi.resolve(:product_name)
+            elsif isa == 'sparc'
+              fact_value = Facter::Resolvers::Solaris::DmiSparc.resolve(:product_name)
+            end
```

Comment:
you can do something like:
`fact_value = if isa == 'i386'
                 Facter::Resolvers::Solaris::Dmi.resolve(:product_name)
            elsif isa == 'sparc'
                Facter::Resolvers::Solaris::DmiSparc.resolve(:product_name)
            end`